docs/docs/file-system-route-api.md

```diff
@@ -0,0 +1,299 @@
+
+```js:title=src/pages/users/[id].js
+function UserPage(props) {
+  const id = props.params.id
+}
+```
+
+```js:title=src/pages/image/[...awsKey].js
+function ProductsPage(props) {
+  const splat = props.params.awsKey
+}
+```
+
+```js:title=src/pages/image/[...].js
+function AppPage(props) {
+  const splat = props.params[‘*’]
+}
+```
+
+## Routing and linking
+
+Gatsby "slugifies" every route that gets created from collection pages. When you want to link to one of those pages, it may not always be clear how to construct the URL from scratch.
+
+To address this issue, Gatsby automatically includes a `gatsbyPath` field on every type used by collection pages. The `gatsbyPath` field must take an argument of the `filePath` it is trying to resolve. This is necessary because it’s possible that one type is used in multiple collection pages.
+
+There are some general syntax requirements when using the `filePath` argument:
+
+- The path must be an absolute path (so starting with a `/`)
+- You must omit the file extension
+- You must omit the `src/pages` prefix
+- Your path must not include `index`
+
+### `gatsbyPath` example
+
+Assume that a `Product` type is used in two pages:
+
+- `src/pages/products/{Product.name}.js`
+- `src/pages/discounts/{Product.name}.js`
+
+If you wanted to link to the `products/{Product.name}` and `discounts/{Product.name}` routes from your home page, you would have a component like this:
+
+```jsx:title=src/pages/index.js
+import React from "react"
+import { Link, graphql } from "gatsby"
+
+export default function HomePage(props) {
+  return (
+    <ul>
+      {props.data.allProduct.map(product => (
+        <li key={product.name}>
+          <Link to={product.productPath}>{product.name}</Link> (<Link to={product.discountPath}>Discount</Link>)
+        </li>
+      ))}
+    </ul>
+  )
+}
+
+export const query = graphql`
+  query {
+    allProduct {
+      name
+      productPath: gatsbyPath(filePath: "/products/{Product.name}")
+      discountPath: gatsbyPath(filePath: "/discounts/{Product.name}")
+    }
+  }
+}
+```
+
+By using [aliasing](/docs/graphql-reference/#aliasing) you can use `gatsbyPath` multiple times.
+
+## Example use cases
+
+Have a look at the [route-api example](https://github.com/gatsbyjs/gatsby/tree/master/examples/route-api/src/pages/products) for more detail.
+
+### Collection route + fallback
+
+By using a combination of a collection route with a client only route, you can create a great experience when a user tries to visit a URL from the collection route that doesn’t exist for the collection item. Consider these two file paths:
+
+- `src/pages/products/{Product.name}.js`
+- `src/pages/products/[name].js`
+
+If the user visits a product that wasn’t built from the data in your site, they will hit the client-only route, which can be used to show the user that the product doesn’t exist or load the data on the client.
```

Comment:
> If the user visits a product that wasn’t built from the data in your site, they will hit the client-only route, **which can be used to show the user that the product doesn’t exist or load the data on the client.**

This part is confusing to me. Could you break it down in a bit more detail? I think I'm not understanding the two options presented. Why would they have some products that are built from the data in their site and some products that aren't?